### PR TITLE
Update fog-core to latest version

### PR DESCRIPTION
**Pull request by pravi**
fog-core 2.x is released and we'd like to use it in debian

### CODE REVIEW

File: vagrant-libvirt.gemspec [hunk 1]
```diff
@@ -22,7 +22,7 @@ Gem::Specification.new do |s|
 
   s.add_runtime_dependency 'fog-libvirt', '>= 0.3.0'
   s.add_runtime_dependency 'nokogiri', '~> 1.6.0'
-  s.add_runtime_dependency 'fog-core', '~> 1.43.0'
+  s.add_runtime_dependency 'fog-core', '~> 2.1.0'
```
Comment (electrofelix) [inline vagrant-libvirt.gemspec]: I'm not 100% sure what the impact of this change would be, so far it looks like it adds a dependency on ruby >= 2.0, but as vagrant 1.8 used ruby 2.2 I'm not sure that is a problem.

**Comment by electrofelix** [comment]
If you wouldn't might pushing an update due to a fix I landed around nokogiri, and I'll look to test it out more over the next few days.

**Comment by pravi** [comment]
@electrofelix rebased and pushed (since fog-core follows SemVer, made dependency less strict to ~> 2.1)

File: vagrant-libvirt.gemspec [hunk 2]
```diff
@@ -21,7 +21,8 @@ Gem::Specification.new do |s|
   s.add_development_dependency "rspec-mocks", "~> 3.5.0"
 
   s.add_runtime_dependency 'fog-libvirt', '>= 0.3.0'
-  s.add_runtime_dependency 'fog-core', '~> 1.43.0'
+  s.add_runtime_dependency 'nokogiri', '~> 1.6.0'
```
[the remark this answers is not in the document]
Comment (pravi) [reply vagrant-libvirt.gemspec]: removed duplicate entry

**Comment by electrofelix** [comment]
Needed to retrigger travis build

**Comment by avollmerhaus** [comment]
Is there any chance that this will make it into a release?
At the moment I'm having trouble because of conflicting dependencies with the vagrant-reload plugin:
```
vagrant plugin install vagrant-reload
Installing the 'vagrant-reload' plugin. This can take a few minutes...
Vagrant failed to properly resolve required dependencies. These
errors can commonly be caused by misconfigured plugin installations
or transient network issues. The reported error is:

conflicting dependencies fog-core (~> 1.43.0) and fog-core (= 2.1.2)
  Activated fog-core-2.1.2
  which does not match conflicting dependency (~> 1.43.0)

  Conflicting dependency chains:
    fog-core (= 2.1.2), 2.1.2 activated

  versus:
    vagrant-libvirt (> 0), 0.0.41 activated, depends on
    fog-core (~> 1.43.0)
```

**Pull request merged**